1.Exploratory_Data_Analysis/scripts/2c.correlation_nomic_and_sc_morphology.py

```diff
@@ -0,0 +1,382 @@
+# tmpdf = tmpdf.corr()
+# # drop sc data columns correlation to sc data columns
+# tmpdf = tmpdf.iloc[0:sc_data_features_agg.shape[1]-2, sc_data_features_agg.shape[1]-2:]
+# tmpdf = tmpdf.reset_index()
+# tmpdf = tmpdf.rename(columns={'index': 'feature'})
+# # save the correlation matrix to a csv
+
+# # melt the correlation matrix
+# corr_matrix_melted = tmpdf.reset_index().melt(id_vars='feature', value_vars=tmpdf.columns[1:])
+# corr_matrix_melted = corr_matrix_melted.rename(columns={'index': 'feature', 'variable': 'feature2', 'value': 'correlation'})
+
+
+# # find the treatment dose for well B13 in sc_data_metadata
+
+# corr_matrix_melted['treatment'] = treatment
+# corr_matrix_melted.to_csv(f'{cell_type}_{treatment}_correlation_matrix.csv')
+# # change corralted values that are abs less than 0.5 to 0
+# corr_matrix_melted.loc[abs(corr_matrix_melted['correlation']) < 0.5, 'correlation'] = 0
+# # remove rows that contain all 0s
+# corr_matrix_melted = corr_matrix_melted.loc[~(corr_matrix_melted==0).all(axis=1)]
+# # remove all columns that contain all 0s
+# corr_matrix_melted = corr_matrix_melted.loc[:, (corr_matrix_melted!=0).any(axis=0)]
+# # set pivot table for corr_matrix_melted
+# corr_matrix_melted
+# # sc_data_features_agg
+# corr_matrix_pivot = corr_matrix_melted.pivot_table(index=['feature'], columns='feature2', values='correlation')
+# # remove rows that contain all 0s
+# corr_matrix_pivot = corr_matrix_pivot.loc[~(corr_matrix_pivot==0).all(axis=1)]
+# # remove all columns that contain all 0s
+# corr_matrix_pivot = corr_matrix_pivot.loc[:, (corr_matrix_pivot!=0).any(axis=0)]
+# # plot the pivot table
+# # plt.figure(figsize=(40, 40))
+# # sns.heatmap(corr_matrix_pivot, cmap='coolwarm', vmin=-1, vmax=1)
+# # plt.show()
+# corr_matrix_pivot['Activin A [NSU]'].unique()
+
+
+# In[ ]:
+
+
+# In[15]:
+
+
+for treatment in merged_data_agg[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].unique():
+    print(treatment)
+    # get the row
+    tmpdf = merged_data_agg.loc[
+        merged_data_agg["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] == treatment
+    ]
+    # get the correlation between the sc data and nomic data for well B13
+    tmpdf = merged_data_agg.drop(
+        ["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"], axis=1
+    )
+    tmpdf = tmpdf.corr()
+    # drop sc data columns correlation to sc data columns
+    tmpdf = tmpdf.iloc[
+        0 : sc_data_features_agg.shape[1], sc_data_features_agg.shape[1] :
+    ]
+    tmpdf = tmpdf.reset_index()
+    tmpdf = tmpdf.rename(columns={"index": "feature"})
+    # save the correlation matrix to a csv
+
+    # # melt the correlation matrix
+    corr_matrix_melted = tmpdf.reset_index().melt(
+        id_vars="feature", value_vars=tmpdf.columns[1:]
+    )
+    corr_matrix_melted = corr_matrix_melted.rename(
+        columns={"index": "feature", "variable": "feature2", "value": "correlation"}
+    )
```

Comment:
All of this can be chained, which makes for a much more pleasant reading experience :)

Can you also think of a more readable name than `tmpdf`?